api/client.go

```diff
@@ -74,28 +76,26 @@ func NewClient(l logger.Logger, conf Config) *Client {
 
 	httpClient := conf.HTTPClient
 	if conf.HTTPClient == nil {
-		t := &http.Transport{
-			Proxy:              http.ProxyFromEnvironment,
-			DisableCompression: false,
-			DisableKeepAlives:  false,
-			DialContext: (&net.Dialer{
-				Timeout:   30 * time.Second,
-				KeepAlive: 30 * time.Second,
-			}).DialContext,
-			MaxIdleConns:        100,
-			IdleConnTimeout:     90 * time.Second,
-			TLSHandshakeTimeout: 30 * time.Second,
-		}
+
+		// use the default transport as it is optimized and configured for http2
+		// and will avoid accidents in the future
+		tr := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
Making a note in case this is important later on - no changes needed.

The diff from the old transport to the default is:

```diff
- TLSHandshakeTimeout:   30 * time.Second,
+ ForceAttemptHTTP2:     true,
+	TLSHandshakeTimeout:   10 * time.Second,
+	ExpectContinueTimeout: 1 * time.Second,
```

and the use of  `defaultTransportDialContext` wrapping `net.Dialer`, but that is equivalent to `.DialContext`.

Reply:
Good callout, thanks!